lib/ohai/plugins/aix/network.rb

```diff
@@ -84,56 +83,53 @@
               netmask = IPAddr.new("255.255.255.255").mask(tmp_prefix.to_i).to_s
             end
 
-            iface[interface][:addresses] ||= Mash.new
-            iface[interface][:addresses][tmp_addr] = { "family" => "inet", "prefixlen" => tmp_prefix }
-            iface[interface][:addresses][tmp_addr][:netmask] = netmask
+            ifaces[int_name][:addresses][tmp_addr] = { "family" => "inet", "prefixlen" => tmp_prefix }
+            ifaces[int_name][:addresses][tmp_addr][:netmask] = netmask
 
             if lin =~ /broadcast\s(\S+)\s/
-              iface[interface][:addresses][tmp_addr][:broadcast] = $1
+              ifaces[int_name][:addresses][tmp_addr][:broadcast] = $1
             end
           elsif lin =~ %r{inet6 ([a-f0-9\:]+)%?(\d*)/?(\d*)?}
             # TODO do we have more properties on inet6 in aix? broadcast
-            iface[interface][:addresses] ||= Mash.new
-            iface[interface][:addresses][$1] = { "family" => "inet6", "zone_index" => $2, "prefixlen" => $3 }
+            ifaces[int_name][:addresses] ||= Mash.new
+            ifaces[int_name][:addresses][$1] = { "family" => "inet6", "zone_index" => $2, "prefixlen" => $3 }
           else
-            # load all key-values, example "tcp_sendspace 131072 tcp_recvspace 131072 rfc1323 1"
+            # add all key value data into the interface mash
+            # for example "tcp_sendspace 131072 tcp_recvspace 131072 rfc1323 1"
+            # has keys tcp_sendspace, tcp_recvspace, and rfc1323
             properties = lin.split
-            n = properties.length / 2 - 1
-            (0..n).each do |i|
-              iface[interface][properties[i * 2]] = properties[(i * 2 + 1)]
+            (properties.size / 2).times do
+              ifaces[int_name][properties.shift] = properties.shift
             end
```

Comment:
```suggestion
            lin.split.each_slice(2) do |key, value|
              ifaces[int_name][key] = value
            end
```

Reply:
Or as a one-liner (didn't test, might not work)

```suggestion
            ifaces[int_name].merge!(lin.split.each_slice(2).to_a.to_h)
```

Reply:
I pulled that in with the line modification done above. The 3 liner is easy on the eyes so I took that one. Thanks for further fixing that up.